chipsec/hal/intel/cpu.py

```diff
@@ -173,8 +173,8 @@ def get_TSEG(self) -> Tuple[int, int, int]:
         if self.cs.is_server():
             # tseg register has base and limit
             tseg_base = self.cs.register.get_list_by_name('8086.MEMMAP_VTD.TSEG').read_field('base', True)[0]
-            tseg_limit = self.cs.register.get_list_by_name('8086.MEMMAP_VTD.TSEG').read_field('limit', True)[0]
-            tseg_limit += 0xFFFFF
+            tseg_limit = self.cs.register.get_list_by_name('8086.MEMMAP_VTD.TSEG').read_field('limit', False)[0]
+            tseg_limit = (tseg_limit << 20) + 0xFFFFF
```

Comment:
I think if we left it the way it was, it keeps the bit shift in the xml, which I prefer. Can you try it and confirm?